Explanation boxes

erler opened the pull request:
Added dismissible explanations to Discussions and On Chain, and a explanation popup to Referendum Vote Locking #297 
<img width="555" alt="Screenshot 2020-02-12 at 11 13 12" src="https://user-images.githubusercontent.com/7072141/74325713-62b35680-4d89-11ea-850c-0f4cc02cc9a1.png">
<img width="547" alt="Screenshot 2020-02-12 at 11 13 19" src="https://user-images.githubusercontent.com/7072141/74325716-63e48380-4d89-11ea-8114-b3b391d2278f.png">
<img width="361" alt="Screenshot 2020-02-12 at 11 12 58" src="https://user-images.githubusercontent.com/7072141/74325725-6646dd80-4d89-11ea-9c48-05c9f99cc06f.png">

Visibility state is stored in localStorage - this can be probably further abstracted if we add more info boxes along the way.

niklabh commented on front-end/src/screens/Discussions/index.tsx:
```diff
@@ -14,6 +15,25 @@ const Discussions = ({ className } : {className?: string}) => {
 	};
 
+	useState(() => {
+		var infoVisible = JSON.parse(localStorage?.getItem('infoVisible')!);
```
instead of setting discussionsInfoVisible onchainInfoVisible in a object then stringifying/parsing it for setting/getting into localstorage you can just do 

localStorage.setItem('discussionsInfo', 'visible')
localStorage.getItem('discussionsInfo') === 'visible'

much cleaner that way

erler replied:
Ah, thanks. That's how I had it first, but wasn't sure if an info object to have all that information in makes sense for later on. Changing it back for now :)

Tbaut replied:
My bad, I told you the object way. Either way isn't super costly.

Tbaut commented on front-end/src/screens/Discussions/index.tsx:
```diff
@@ -14,6 +15,17 @@ const Discussions = ({ className } : {className?: string}) => {
 
+	const infoBoxVisible = localStorage.getItem('discussionsInfo');
+	if (infoBoxVisible === null) {
+		localStorage.setItem('discussionsInfo', 'true');
```
The name `discussionInfo` doen't ask a quesion that can be answered by true/false. It'd be preferable to keep the Visisble, hence `discussionInfoVisible`.

Tbaut commented on front-end/src/screens/Discussions/index.tsx:
```diff
@@ -14,6 +15,17 @@ const Discussions = ({ className } : {className?: string}) => {
+	}
+	const [infoVisible, setInfoVisible] = useState(infoBoxVisible);
+
+	const handleDismiss = () => {
```
same, there're a lot of things we can dismiss. We should be more specific and use e.g `handleInfoBoxDismiss`

Tbaut replied:
you can forget about this comment if the `handleDismiss` is inside InfoBox, because it because obvious then.

Tbaut commented on front-end/src/screens/Discussions/index.tsx:
```diff
@@ -14,6 +15,17 @@ const Discussions = ({ className } : {className?: string}) => {
+		localStorage.setItem('discussionsInfoVisible', 'false');
+		setInfoVisible(localStorage.getItem('discussionsInfoVisible'));
+	};
+
```
As said in the first review, please move this inside the InfoBox component.

Tbaut commented on front-end/src/screens/OnChain/index.tsx:
```diff
@@ -1,13 +1,25 @@
+		localStorage.setItem('onchainInfoVisible', 'false');
+		setInfoVisible(localStorage.getItem('onchainInfoVisible'));
+	};
+
```
Seme here, this code doesn't need to be duplicated :)

erler commented:
My apologies - I scanned through the inline comments and oversaw your main comment :/

Moved the logic to the InfoBox component, so Discussions and OnChain screens look much cleaner now.

Tbaut commented on front-end/src/ui-components/InfoBox.tsx:
```diff
@@ -0,0 +1,69 @@
+}
+
+const InfoBox = ({ children, className, content, dismissable, name, title }: Props) => {
+	const infoBoxVisible = localStorage.getItem(name);
```
Please don't forget my first comment, adding `Visible` to the name help clarifying what this does.
```suggestion
	const localStorageName = name + 'Visible'
	const infoBoxVisible = localStorage.getItem(localStorageName);
```

Tbaut commented on front-end/src/ui-components/InfoBox.tsx:
```diff
@@ -0,0 +1,69 @@
+const InfoBox = ({ children, className, content, dismissable, name, title }: Props) => {
+	const infoBoxVisible = localStorage.getItem(name);
+	if (infoBoxVisible === null) {
+		localStorage.setItem(name, 'true');
```
```suggestion
		localStorage.setItem(localStorageName, 'true');
```

Tbaut commented on front-end/src/ui-components/InfoBox.tsx:
```diff
@@ -0,0 +1,69 @@
+		localStorage.setItem(name, 'true');
+	}
+
+	const [infoVisible, setInfoVisible] = useState(JSON.parse(infoBoxVisible!));
```
Do you need this JSON.parse?

erler replied:
Well, I want the boolean value instead of a string

Tbaut replied:
ok then `infoBoxVisible === 'true'` should do the job and would be much easier to read.

Tbaut commented:
Also I think the font size we're using are too small for dropdown and help info. The difference with other fonts it too big. let's make that at least 1.2rem (that's the size of the chain infos)
![image](https://user-images.githubusercontent.com/33178835/74358230-6793fc80-4dc1-11ea-8232-892c3112a529.png)

erler commented:
**Styling**
Grey boxes work I think. Moved the h1 out of the column, so the InfoBox is aligned top of the column. On On Chain that means not aligned with a card, but a subheading. I guess that makes sense as the info is about the entire page, but let me know if you think it looks off.

Fixed the stackable grid on mobile and tablet, so the info comes first. 

Adjusted the tooltip font size (also on Sign Up where we're at it).
<img width="1275" alt="Screenshot 2020-02-12 at 19 57 39" src="https://user-images.githubusercontent.com/7072141/74374313-73da8280-4dde-11ea-85c7-46bb44bf342f.png">
<img width="1273" alt="Screenshot 2020-02-12 at 19 59 22" src="https://user-images.githubusercontent.com/7072141/74374316-763cdc80-4dde-11ea-9379-d3dcb9c7ab54.png">
<img width="1275" alt="Screenshot 2020-02-12 at 20 01 14" src="https://user-images.githubusercontent.com/7072141/74374321-776e0980-4dde-11ea-8deb-051cc691b869.png">

**Functionality**
Initialisation bug is fixed. Thanks for pointing out! Used your suggestions for naming and added cursor pointer on cross hover.

Tbaut commented:
Also the linting job is complaining (all those red crosses)
> $ yarn lint
yarn run v1.22.0
$ npx eslint ./src/ --ext .js,.jsx,.jsx,.ts,.tsx
/builds/parity/polkassembly/front-end/src/ui-components/InfoBox.tsx
  15:43  error    Missing semicolon            semi
  18:1   warning  Trailing spaces not allowed  no-trailing-spaces
✖ 2 problems (1 error, 1 warning)
  1 error and 1 warning potentially fixable with the `--fix` option.

erler commented:
Fixed, sorry.
